crates/lang/src/reflect/trait_def/registry.rs

```diff
@@ -61,3 +67,30 @@ where
 {
     type Env = E;
 }
+
+impl<E> ToAccountId<E> for TraitDefinitionRegistry<E>
+where
+    E: ink_env::Environment,
+{
+    #[cold]
+    fn to_account_id(&self) -> E::AccountId {
+        /// We enforce linking errors in case this is ever actually called.
+        /// These linker errors are properly resolved by the cargo-contract tool.
+        extern "C" {
+            fn _ink_compilation_error() -> !;
+        }
+        unsafe { _ink_compilation_error() }
+    }
```

Comment:
Is this still needed?

Reply:
Yep, if someone will try to call `to_account_id` from `TraitDefinitionRegistry` we want to throw an error during compilation